src/checkbox/Checkbox.ts

```diff
@@ -89,31 +107,35 @@ export default class Checkbox extends CheckboxBase<CheckboxProperties> {
 			required ? css.required : null
 		];
 
-		const checkbox = v('div', { classes: this.classes(css.inputWrapper) }, [
-			v('input', {
-				bind: this,
-				classes: this.classes(css.input),
-				checked,
-				'aria-describedby': describedBy,
-				disabled,
-				'aria-invalid': invalid + '',
-				name,
-				readOnly,
-				'aria-readonly': readOnly ? 'true' : null,
-				required,
-				type: 'checkbox',
-				value,
-				onblur: this._onBlur,
-				onchange: this._onChange,
-				onclick: this._onClick,
-				onfocus: this._onFocus,
-				onmousedown: this._onMouseDown,
-				onmouseup: this._onMouseUp,
-				ontouchstart: this._onTouchStart,
-				ontouchend: this._onTouchEnd,
-				ontouchcancel: this._onTouchCancel
-			})
-		]);
+		const children = [
+			mode === Mode.toggle && offLabel ? v('div', { classes: this.classes(css.offLabel) }, [ offLabel ]) : null,
+			v('div', { classes: this.classes(css.inputWrapper) }, [
+				v('input', {
+					bind: this,
+					classes: this.classes(css.input),
+					checked,
+					'aria-describedby': describedBy,
+					disabled,
+					'aria-invalid': invalid + '',
+					name,
+					readOnly,
+					'aria-readonly': readOnly ? 'true' : null,
+					required,
+					type: 'checkbox',
+					value,
+					onblur: this._onBlur,
+					onchange: this._onChange,
+					onclick: this._onClick,
+					onfocus: this._onFocus,
+					onmousedown: this._onMouseDown,
+					onmouseup: this._onMouseUp,
+					ontouchstart: this._onTouchStart,
+					ontouchend: this._onTouchEnd,
+					ontouchcancel: this._onTouchCancel
+				})
+			]),
+			mode === Mode.toggle && onLabel ? v('div', { classes: this.classes(css.onLabel) }, [ onLabel ]) : null
```

Comment:
I think it makes more sense to have `onLabel` and `offLabel` inside `div.inputWrapper` so they can be positioned relative to the control indicator (part of the reason I created a separate wrapping div). Otherwise it'll be hard to style them correctly, what with variable before/after label text and all.